Loader/Loader/Loader.server.lua

```diff
@@ -9,88 +9,130 @@
-else
-	_G["__Adonis_MUTEX"] = script:GetFullName()
-	
-	local model = script.Parent.Parent
-	local config = model.Config
-	local core = model.Loader
+end
+
+--// Root Folder Instances
+local Model = script.Parent.Parent
+local Config = Model.Config
+local Core = Model.Loader
+local Backup = Model:Clone()
+local OrigName = Model.Name
+
+--// Core Instances
+local Dropper = Core.Dropper
+local Loader = Core.Loader
+local Runner = script
+
+--// Get Configuration Instances
+local Settings = Config.Settings
+local Plugins = Config.Plugins
+local Themes = Config.Themes
+
+--// Define valid Plugin Types
+local PluginTypes = {"server:";"server-";"client-";"client:"}
+
+local Data = {
+	Settings = {};
+	Descriptions = {};
+	Order = {};
+	ServerPlugins = {};
+	ClientPlugins = {};
+	Themes = {};
+	StartTime = tick();
 	
-	local dropper = core.Dropper
-	local loader = core.Loader
-	local runner = script
+	Model = Model;
+	Config = Config;
+	Core = Core;
 	
-	local settings = config.Settings
-	local plugins = config.Plugins
-	local themes = config.Themes
+	Loader = Loader;
+	Dropper = Dropper;
+	Runner = Runner;
+
+		ModuleID = tonumber('23735'..'01710'); --// Trying to break existing (unupdatable) malicious plugins that replace the ModuleID from studio on insertion
+	LoaderID = tonumber('23735'..'05175');
+}
+
+local LoadPlugins = function()
+	for _, Plugin in pairs(Plugins:GetChildren()) do
+		local Type = tostring(Plugin.Name):lower():sub(1,7)
+		if Plugin:IsA("ModuleScript") and table.find(PluginTypes, Type) then
+			if Type:match("client") then
+				table.insert(Data.ClientPlugins, Plugin)
+			elseif Type:match("server") then
+				table.insert(Data.ServerPlugins, Plugin)
+			else
+				warn("Unknown Plugin Type for: "..tostring(Plugin))
+			end
+		end
+	end
+end
+
+local LoadThemes = function()
+	for _, Theme in pairs(Themes:GetChildren()) do
+		table.insert(Data.Themes, Theme)
+	end
+end
+
+local Load = function()
+	warn("Loading...")
 	
-	local backup = model:Clone()
+	_G.__Adonis_MUTEX = script:GetFullName()
+	script:Destroy()
 	
-	local data = {
-		Settings = {};
-		Descriptions = {};
-		ServerPlugins = {};
-		ClientPlugins = {};
-		Themes = {};
-		
-		Model = model;
-		Config = config;
-		Core = core;
-		
-		Loader = loader;
-		Dopper = dropper;
-		Runner = runner;
-		
-		ModuleID = tonumber('23735'..'01710'); --// Trying to break existing (unupdatable) malicious plugins that replace the ModuleID from studio on insertion
-		LoaderID = tonumber('23735'..'05175');
-		
-		DebugMode = true;
-	}
+	Model.Name = math.random()
 	
 	--// Init
-	script:Destroy()
-	model.Name = math.random()
-	local moduleId = data.ModuleID
-	local a,setTab = pcall(require,settings)
-	if not a then
-		warn'::Adonis:: Settings module errored while loading; Using defaults;'
-		setTab = {}
+	local ModuleID = Data.ModuleID
+	local success, settings = pcall(require, Settings)
+
+	if not success then
+		warn("Settings module failed to load; Using defaults;")
+		settings = {}
```

Comment:
You have just reimplemented `pcall` to output an error message if the function has failed, yet you have decided to output a `warn` manually?

Reply:
Can you resubmit this, i can't change it because it's "outdated"